src/LEGO.AsyncAPI.Readers/YamlConverter.cs

```diff
@@ -47,22 +47,41 @@ public static JsonNode ToJsonNode(this YamlNode yaml)
             };
         }
 
-        private static JsonValue ToJsonValue(this YamlScalarNode yaml)
+        public static JsonValue ToJsonValue(this YamlScalarNode yaml)
         {
+            string value = yaml.Value;
+
             switch (yaml.Style)
             {
                 case ScalarStyle.Plain:
-                    return decimal.TryParse(yaml.Value, NumberStyles.Float, CultureInfo.InvariantCulture, out var d)
```

Comment:
`decimal` was a weird choice for capturing all number, is most serializers I have worked with they will default to a `double`. This code now correctly supports parsing to `int`, `double`, ` DateTime`, `bool`, and `string.

Reply:
Decimal was ultimately chosen as double looses precision.
This might be fine in many instances with calculating certain things, but not when representing numbers
input should be equal to output, with max precision.

double looses precision with certain conversions.
A few examples
`1.2345678901234567890123456789E25`
`12345678901234567890.0123456789`